### ISSUE TITLE
Home App does not sync status 

### ISSUE THREAD

**Issue by kamaruszamanjufri** [bwp91/homebridge-ewelink-sonoff#14]
**What issue do you have?**
...
Home App does not sync status when the sonoff device is power off. It is still shows previous status.

**Which version of the package are you using?**
...
2.9.4
**Please paste any relevant logs below**
...
[7/27/2020, 9:57:46 PM] [eWeLink] [Hall Fan 2] found in eWeLink but LAN mode unsupported.
[7/27/2020, 9:57:46 PM] [eWeLink] [Hall Fan 2] will not be refreshed as it has been reported offline.
[7/27/2020, 9:58:31 PM] [eWeLink] [Hall Fan 2] it is currently offline.

**Comment by bwp91** [comment]
I will get back to this... just want to get the fan sorted first as per #13

**Comment by kamaruszamanjufri** [comment]
Noted.

**Comment by bwp91** [comment]
> Noted.

Thank you for noting this.
Getting back to this now as I think the fan is pretty much sorted?
Regarding offline devices - 
First I would like to ask what your expected behaviour would be from the Home app if a device goes offline.
It is impossible to query the status of a device if it is offline, as it is offline, it can't "give feedback" on its current status.
In the last update to this plugin, I have added a few feature - that once a device comes back online then the plugin will request a device update and hence update the status in Homebridge.
But there are always limitations when working between different platforms ie Home App (HomeKit) <-> Homebridge <-> CoolKit server <-> eWeLink <-> Your devices on your network.
If a device goes offline there can be a delay even between it going offline and it appearing offline in the eWeLink app. This plugin will get a notification from eWeLink that the device is offline, but this could take a few minutes, depending on how quickly eWeLink updates the status.
So to recap my questions are:
1. is it regular for a sonoff device to go offline?
2. how would you like home app to react?
3. what could i do further in this plugin to improve the user experience for devices that go offline?

Thanks again for your patience with the fan!
Ben

**Comment by kamaruszamanjufri** [comment]
Hi Ben,

Below are my Favourite Accessories.
Some of them are in off line state and not reflected correctly in Home App
such as  the Hall Fan 1, Kitchen Water Pump, Ventilation etc.

If I want to double confirm the state, I usually click the icon and the status will show “No Response”.

[A close up of a computer  Description automatically generated]

So to recap my questions are:

  1.  is it regular for a sonoff device to go offline?
It is our practice to turn off the switch physically before going out. Internet connectivity issues is seldom.

  1.  how would you like home app to react?
Displayed it as offline. I know the difficulties of knowing the actual state.

  1.  what could i do further in this plugin to improve the user experience for devices that go offline?
I hope that whenever the Home App is at the forefront it can query the updated status from the ewelink.
For the SonOff TH10 although I split the display to two, I can’t change the name to correctly display the temperature and the humidity.

BTW, I got this wrong status for the Water Sensor “Attic, Water Pump”. It triggered in Home App although in Aqara it shows normal.

Regards,
kamaruszaman jufri
kamarus@gmail.com

**Comment by bwp91** [comment]
> If I want to double confirm the state, I usually click the icon and the status will show “No Response”.
> Displayed it as offline. I know the difficulties of knowing the actual state.
In the most recent update I have made modifications so that if you a click on a device to update status, it will show as "No Response" is the ewelink response says the device is offline. Please try it out.
Once a device comes back online then it will refresh its status with ewlink.
I hope this is sufficient as there isn't much more I can do with offline devices.

> For the SonOff TH10 although I split the display to two, I can’t change the name to correctly display the temperature and the humidity.
You can't change the name through the Home App? How does this affect the temperature and the humidity readings?

> I got this wrong status for the Water Sensor “Attic, Water Pump”. It triggered in Home App although in Aqara it shows normal.
Do you use this device through the Sonoff bridge? Otherwise I don't think this has anything to do with this sonoff plugin.

Thanks
Ben

**Comment by kamaruszamanjufri** [comment]
Hi Ben,


If I want to double confirm the state, I usually click the icon and the status will show “No Response”.
Displayed it as offline. I know the difficulties of knowing the actual state.

In the most recent update I have made modifications so that if you a click on a device to update status, it will show as "No Response" is the ewelink response says the device is offline. Please try it out.
Once a device comes back online then it will refresh its status with ewlink.
I hope this is sufficient as there isn't much more I can do with offline devices.

  *   Okay. It’s working.

For the SonOff TH10 although I split the display to two, I can’t change the name to correctly display the temperature and the humidity.

You can't change the name through the Home App? How does this affect the temperature and the humidity readings?

  *   My mistake. I can change the wording in Home App.

I got this wrong status for the Water Sensor “Attic, Water Pump”. It triggered in Home App although in Aqara it shows normal.

Do you use this device through the Sonoff bridge? Otherwise I don't think this has anything to do with this sonoff plugin.

  *   Just to know if you have encounter the same.

Regards,
kamaruszaman jufri
kamarus@gmail.com

**Comment by bwp91** [comment]
> Okay. It’s working. 
> My mistake. I can change the wording in Home App.

Good to hear!
> Just to know if you have encounter the same.

Nope I don't have any sensors like this.

**Comment by bwp91** [comment]
Hi,
With regards to the initial issue "Home App does not sync status" I hope the solution I have implemented and explained above is sufficient.
I would like to close this issue if you are happy for me to?
And of course if you encounter further issues please do let me know.
Thanks!

**Comment by kamaruszamanjufri** [comment]
Ok
